src/mlpack/tests/adaboost_test.cpp

```diff
@@ -62,7 +65,9 @@ BOOST_AUTO_TEST_CASE(HammingLossBoundIris)
       countError++;
   double hammingLoss = (double) countError / labels.n_cols;
 
-  BOOST_REQUIRE_LE(hammingLoss, a.ZtProduct());
+  // Check that ztProduct is finite.
+  BOOST_REQUIRE_EQUAL(boost::math::fpclassify(ztProduct), FP_NORMAL);
```

Comment:
Probably easier to use `std::isfinite()` here, then we don't need to bring in the Boost libraries. :+1:

Reply:
Yes, actually I saw everything in testcases is made with `boost` so decided to go with `boost`. Later on noticed the unwanted overhead due to inclusion of boost library as `Std` is already included .Sorry for that.